io/blob.go

```diff
@@ -119,6 +121,64 @@ func (bfs *blobFileIO) WriteFile(name string, content []byte) error {
 	return bfs.Bucket.WriteAll(bfs.ctx, name, content, nil)
 }
 
+// ListObjects implements the ListIO interface for blob-based file systems
+func (bfs *blobFileIO) ListObjects(prefix string, fn func(path string, info fs.FileInfo) error) error {
```

Comment:
this is closer to a `Walk` method than a `List` method since it is calling a function on each object.